src/build/functions/edge.ts

```diff
@@ -174,10 +184,12 @@ export const createEdgeHandlers = async (ctx: PluginContext) => {
   ]
   await Promise.all(nextDefinitions.map((def) => createEdgeHandler(ctx, def)))
 
-  const netlifyDefinitions = nextDefinitions.flatMap((def) => buildHandlerDefinition(ctx, def))
-  const netlifyManifest: Manifest = {
-    version: 1,
-    functions: netlifyDefinitions,
+  if (!ctx.useFrameworksAPI) {
+    const netlifyDefinitions = nextDefinitions.flatMap((def) => buildHandlerDefinition(ctx, def))
+    const netlifyManifest: Manifest = {
+      version: 1,
+      functions: netlifyDefinitions,
+    }
+    await writeEdgeManifest(ctx, netlifyManifest)
```

Comment:
this doesn't feel great to use conditional check like that in 2 separate places that relate to ~same thing (wether ~config is declared in manifest.json file or inline in function) - it feels brittle and also it feels hard to read

there is some simple abstraction that can be added to maybe increase readability - like introduce `shouldGenerateManifest` and `shouldUseInlineConfig` and following those make it easier to understand what happens when, but for now I wanted to do least code changes possible to maybe see if someone has nice idea how to best deal with those conditional code paths this isn't over the top overengineered but is still readable and easy to follow?